OpenTrons/opentrons-api
# refactor(app): update robot software update flow for unboxing

# Overview
[design]
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Release%3A-Opentrons-Flex-Touchscreen-Designs?type=design&node-id=7147-91378&t=D58715chNPzn7LFs-0

update the robot software update flow to align with hi-fi design.



# Test Plan
edit `config.json` 
```json
"unfinishedUnboxingFlowRoute": "/welcome"
```
restart a robot/dev kit
command `/sbin/reboot`

1. set up network flow
2. 
3. go to software update
This process will take some time and it depends on your network situation (at office around 3-4 min). Maybe the progress bar's animation on ODD is laggy. 
If the odd app shows, the robot software is updated already something.
change currentVersion in`app/src/redux/buildroot/selectors.ts` to the following.

```ts
const currentVersion = '0.0.0'
```

4. after updating the software, the app should show the rename
unfinishedUnboxingFlowRoute should be update to the following.
```json
"unfinishedUnboxingFlowRoute": "/robot-settings/rename-robot"
```



# Changelog
- update components that are related to robot software update


# Review requests



# Risk assessment
low
